### PR TITLE
GX Render - React Native iOS example

### PR DESCRIPTION
**Pull request by CinthyaRiverosGlb**
# SuperApp Example
This document explains how to develop and integrate the functionality that provides the API for access to the Mini Apps Center, as well as the API for managing their cache, based on the MiniAppCaller example.

## How to run the react native example

Install the node modules and pods folders on the root folder ExampleSuperApp

npm install && cd ios && pod install && pod update && cd ..

###### _note: If you don't have npm installed please follow the following instructions: [Install Node and npm](https://www.jcchouinard.com/install-node-and-npm/)_

To run de metro (react native server)
 
npx react-native start

                        ▒▒▓▓▓▓▒▒
                     ▒▓▓▓▒▒░░▒▒▓▓▓▒
                  ▒▓▓▓▓░░░▒▒▒▒░░░▓▓▓▓▒
                 ▓▓▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▓▓
                 ▓▓░░░░░▒▓▓▓▓▓▓▒░░░░░▓▓
                 ▓▓░░▓▓▒░░░▒▒░░░▒▓▒░░▓▓
                 ▓▓░░▓▓▓▓▓▒▒▒▒▓▓▓▓▒░░▓▓
                 ▓▓░░▓▓▓▓▓▓▓▓▓▓▓▓▓▒░░▓▓
                 ▓▓▒░░▒▒▓▓▓▓▓▓▓▓▒░░░▒▓▓
                  ▒▓▓▓▒░░░▒▓▓▒░░░▒▓▓▓▒
                     ▒▓▓▓▒░░░░▒▓▓▓▒
                        ▒▒▓▓▓▓▒▒

                Welcome to Metro v0.73.9
              Fast - Scalable - Integrated

r - reload the app
d - open developer menu
i - run on iOS
a - run on Android

## Execute ios project 
Open ExampleSuperApp/ios/ExampleSuperApp.xcworkspace or select the option i - run on iOS from Metro  

## FAQ:
image
Solution: Add to Build Settings > Framework Search Paths your path to 

`gx-super-app/iOS/GeneXus\ Frameworks/`

### CODE REVIEW

File: Android/MiniAppCaller/app/src/main/java/com/genexus/superapps/bankx/ui/screens/main/MiniAppListHomeContent.kt [hunk 1]
```diff
@@ -73,7 +73,7 @@ fun MiniAppListHomeContent(model: MainViewModel) {
             }
         }
 
-        if (BuildConfig.FLAVOR == Flavor.SANDBOX && Services.SuperApps.Prototyping.isEnabled) {
+        if (BuildConfig.FLAVOR == Flavor.SANDBOX && Services.SuperApps.isSandboxEnvironment()) {
```
Comment (jotapeg) [inline Android/MiniAppCaller/app/src/main/java/com/genexus/superapps/bankx/ui/screens/main/MiniAppListHomeContent.kt:76]: Este es uno de los cambios que no puede bajar

File: Android/MiniAppCaller/app/src/main/java/com/genexus/superapps/bankx/viewmodel/main/MainViewModel.kt [hunk 2]
```diff
@@ -58,7 +58,7 @@ class MainViewModel : ViewModel() {
 
     @SuppressLint("MissingPermission")
     suspend fun loadSandbox(context: Context) {
-        val task = Services.SuperApps.Prototyping.loadSandbox()
+        val task = Services.SuperApps.loadSandbox()
```
Comment (jotapeg) [inline Android/MiniAppCaller/app/src/main/java/com/genexus/superapps/bankx/viewmodel/main/MainViewModel.kt:61]: Este es otro que tampoco puede bajar

File: Android/MiniAppCaller/build.gradle [hunk 3]
```diff
@@ -1,7 +1,7 @@
 buildscript {
     ext {
         compose_version = '1.3.0-beta01'
-        fc_version = "1.7-SNAPSHOT"
+        fc_version = "1.6-SNAPSHOT"
```
Comment (jotapeg) [inline Android/MiniAppCaller/build.gradle:4]: Y este es otro que tampoco baja

File: Android/Logging.md [hunk 4]
```diff
@@ -1,8 +0,0 @@
-# Enable Mini App Logging for Android
-
-The Super App may display Mini App logging messages. 
-To do this, enable Mini-Apps login in the Super App, as shown in [Banking app example](/Android/MiniAppCaller/app/src/main/java/com/genexus/superapps/bankx/application/BankingApplication.kt#L39) 
-
-```kotlin
-Services.Log.setLevel(LogLevel.DEBUG)
-```
```
Comment (jotapeg) [inline Android/Logging.md:8]: ¿Por qué se borró esto?

**Review by jotapeg — DISMISSED** [review]

File: CreateSuperApp.md [hunk 5]
```diff
@@ -3,20 +3,20 @@
 Steps to take to turn an App into a Super App:
 
 1. Integrate the GeneXus Super App Render (aka Flexible Client). It's the main component and is in charge of doing the **Mini App render**
-2. Provide the communication interface in the Super App, which will be called from the integrated Mini Apps
+2. Integrate the **module to access the Mini Apps Center**, so as to get the Mini Apps and be able to load a Mini App
+3. Design the UX for the discovery of those Mini Apps
+4. Provide the communication interface in the Super App, which will be called from the integrated Mini Apps
```
Comment (jotapeg) [inline CreateSuperApp.md:8]: Este archivo tampoco entendí por qué cambió y se borró lo de sandbox al final

File: ReactNative/ExampleSuperApp/ios/ExampleSuperApp.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -0,0 +1,856 @@
...
+		C9E747992A35504500D3F898 /* GXStandardClasses.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = GXStandardClasses.framework; path = "../../../../../Library/Developer/Xcode/DerivedData/GXFrameworks-behydqtgtzwwlnamlgkbycqxuaem/Build/Products/Debug-iphonesimulator/GXStandardClasses.framework"; sourceTree = "<group>"; };
+		C9E7479A2A35504500D3F898 /* GXUIApplication.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = GXUIApplication.framework; path = "../../../../../Library/Developer/Xcode/DerivedData/GXFrameworks-behydqtgtzwwlnamlgkbycqxuaem/Build/Products/Debug-iphonesimulator/GXUIApplication.framework"; sourceTree = "<group>"; };
+		C9E7479B2A35504500D3F898 /* GXFoundation.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = GXFoundation.framework; path = "../../../../../Library/Developer/Xcode/DerivedData/GXFrameworks-behydqtgtzwwlnamlgkbycqxuaem/Build/Products/Debug-iphonesimulator/GXFoundation.framework"; sourceTree = "<group>"; };
```
Comment (finthamoussu) [inline ReactNative/ExampleSuperApp/ios/ExampleSuperApp.xcodeproj/project.pbxproj:117]: These relative paths are not correct. Should point to [iOS/GeneXus Frameworks](https://github.com/genexus-books/gx-super-app/tree/main/iOS/GeneXus%20Frameworks)

File: ReactNative/ExampleSuperApp/ios/GeneXus Frameworks/README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
Comment (finthamoussu) [inline ReactNative/ExampleSuperApp/ios/GeneXus Frameworks/README.md:1]: This folder is not needed, point to [iOS/GeneXus Frameworks](https://github.com/genexus-books/gx-super-app/tree/main/iOS/GeneXus%20Frameworks)

**Review by finthamoussu — CHANGES_REQUESTED** [review]

**Review by finthamoussu — APPROVED** [review]
Updated dependencies to use SPM, the rest LGTM.

**Pull request merged**